bundles/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java

```diff
@@ -0,0 +1,485 @@
+        if (command instanceof RefreshType) {
+            updateNow();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SmhiConfiguration.class);
+
+        connection = new SmhiConnector(httpClient);
+
+        // Check which channel groups are selected in the config.
+        List<Channel> channels = new ArrayList<>();
+        channels.addAll(createChannels());
+        updateThing(editThing().withChannels(channels).build());
+
+        startPolling();
+        updateNow();
+    }
+
+    /**
+     * Start polling for updated weather forecast.
+     */
+    private synchronized void startPolling() {
+        logger.debug("Start polling");
+        forecastUpdater = scheduler.scheduleWithFixedDelay(this::waitForForecast, 1, 1, TimeUnit.MINUTES);
+    }
+
+    /**
+     * Cancels all jobs.
+     */
+    private synchronized void cancelPolling() {
+        logger.debug("Cancelling polling");
+        Future<?> localRef = forecastUpdater;
+        if (localRef != null) {
+            localRef.cancel(false);
+        }
+        localRef = instantUpdate;
+        if (localRef != null) {
+            localRef.cancel(false);
+        }
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        if (!isModifyingCurrentConfig(configurationParameters)) {
+            return;
+        }
+
+        validateConfigurationParameters(configurationParameters);
+
+        Configuration configuration = editConfiguration();
+        for (Map.Entry<String, Object> configurationParameter : configurationParameters.entrySet()) {
+            configuration.put(configurationParameter.getKey(), configurationParameter.getValue());
+        }
+
+        if (isInitialized()) {
+            updateConfiguration(configuration);
+            config = configuration.as(SmhiConfiguration.class);
+            updateThing(editThing().withChannels(createChannels()).build());
+            if (!isPollingStarted()) {
+                startPolling();
+            }
+            updateNow();
+        } else {
+            // persist new configuration and notify Thing Manager
+            updateConfiguration(configuration);
+            ThingHandlerCallback callback = getCallback();
+            if (callback != null) {
+                callback.configurationUpdated(this.getThing());
+            } else {
+                logger.warn("Handler {} tried updating its configuration although the handler was already disposed.",
+                        this.getClass().getSimpleName());
+            }
+        }
+    }
+
+    /**
+     * Update channels with new forecast data.
+     * 
+     * @param timeSeries A {@link TimeSeries} object containing forecasts.
+     */
+    private void updateChannels(TimeSeries timeSeries) {
+        // Loop through hourly forecasts and update those available
+        for (int i = 0; i < 25; i++) {
```

Comment:
How many hours are in a day, does it really include hour 0 and 24? That is 25 data points, just checking :-)

Reply:
Hour 0 is the current hour, and then you can add channels for the next hour (1) up to 24 hours from now :)